processing/src/main/java/io/druid/granularity/QueryGranularity.java

```diff
@@ -108,4 +110,22 @@ else if(o instanceof Number)
     }
     throw new IAE("Cannot convert [%s] to QueryGranularity", o.getClass());
   }
+
+  //simple merge strategy on query granularity that checks if all are equal or else
+  //returns null. this can be improved in future but is good enough for most use-cases.
+  public static QueryGranularity mergeQueryGranularities(List<QueryGranularity> toMerge)
```

Comment:
is it possible here instead to return a map of granularities? Granularity <-> Interval